shared/fs/folder/drop-target.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import * as React from 'react'
+import {Box, Icon} from '../../common-adapters'
+import * as Styles from '../../styles'
+import {isMobile} from '../../constants/platform'
+
+type State = {|
+  showDropOverlay: boolean,
+|}
+
+type Props = {
+  children?: React.Node,
+  onAttach: Array<string> => void,
+}
+
+const DropOverlay = ({onDragLeave, onDrop}) => (
+  <Box style={styles.dropOverlayStyle} onDragLeave={onDragLeave} onDrop={onDrop}>
+    <Icon type="icon-dropping-file-48" />
+  </Box>
+)
+
+class DropTarget extends React.PureComponent<Props, State> {
+  state = {showDropOverlay: false}
+
+  _onDrop = e => {
+    if (!this._validDrag(e)) {
+      return
+    }
+    const fileList = e.dataTransfer.files
+    const paths = fileList.length ? Array.prototype.map.call(fileList, f => f.path) : []
```

Comment:
Could do `fileList.map(f=>f.path)`; also, is `e.dataTransfer.files` ever null? If not, could just do `const paths = e.dataTransfer.files.map(f => f.path)` here instead of checking length.

Reply:
Added a comment explaining this. files is a FileList - not an array, thus a .map won't work.